### DISCUSSION TITLE
Template Announcement

### DISCUSSION THREAD

**Discussion by leaanthony** [wailsapp/wails#825] [Releases]
This is a place to announce new templates.

**Comment by marktohark** [comment]
**WailsCleanVue2Template**
**About：**
it's a clean vue 2. exclude any plugin, something is like vuex, vue-router...
it mean that you can custom plugin by yourself.

**github**
https://github.com/marktohark/WailsCleanVue2Template.git

**Reply by leaanthony** [reply]
Amazing. Looks like the default README is for the alpha veraion. I'll fix it then do a PR for you. Also consider doing a PR for the community templates page on the website 👍

**Reply by marktohark** [reply]
ok, thanks.
in fact, I use beta.3 to generate this template.

**Comment by misitebao** [comment]
**Wails Template Vue**

Description:
Partially optimize the project generated by Vue CLI to provide the most basic single-page routing and internationalization functions.

Link:
[https://github.com/misitebao/wails-template-vue](https://github.com/misitebao/wails-template-vue)

Usage:

```shell
wails init -n Wails-Vue -t https://github.com/misitebao/wails-template-vue
```

**Comment by raitonoberu** [comment]
**Wails Template Svelte**

Based on the [default Svelte template](https://github.com/sveltejs/template). It works fine for me, please let me know if any changes are required!

Link:
https://github.com/raitonoberu/wails-svelte-template

**Comment by BillBuilt** [comment]
**Wails Vite Svelte + TailwindCSS**

These were created by using the Vite CLI tool and choosing Svelte. Only lightly tested on a Mac but they're working for me (so far ;) ). Please let me know if you have any issues.

https://github.com/BillBuilt/wails-vite-svelte-template
https://github.com/BillBuilt/wails-vite-svelte-tailwind-template

**Comment by Sloaix** [comment]
**Wails + React + Typescript + Vite + Tailwind CSS (for Wails v2)**

About：
- Wails + React + Typescript + Vite + Tailwind CSS 
- Support hot-reload (Run ```wails dev``` directly)

Link
https://github.com/Sloaix/wails-vite-react-tailwind-ts-template

**Comment by a-atalla** [comment]
## Vite react template
- Wails 2.0 template with just Vite, React and typescript 
- frontend hot reload works with  `wails dev` directly 
[wails-react-vite-template](https://github.com/a-atalla/wails-react-vite-template)

**Comment by vixalie** [comment]
# Vite react tailwind template

- Wails + Vite + React + Typescript.
- Combines Tailwind CSS and Emotion, using Twin.macro.
- Some utility libraries.
- Support hot reload with `wails dev`.

Link:
[wails-vite-react-tailwind-macro-template](https://github.com/vixalie/wails-vite-react-tailwind-macro-template.git)

**Comment by KiddoV** [comment]
Current ``Vanilla Template``  is still quite... "not really" a ``vanilla`` JS. So I made this repo.
I hope this help someone who love to ``build thing from scratch`` like me without using ``npm`` and other frameworks.

Link:
[wails-pure-js-template](https://github.com/KiddoV/wails-pure-js-template)

Let me know if anything that need to be... fix!
Thanks,

**Reply by leaanthony** [reply]
Nice one! There's always the "plain" template.

**Reply by leaanthony** [reply]
Feel free to add it to the community templates page 👍

**Reply by KiddoV** [reply]
Thanks! Will do

**Comment by rnice01** [comment]
### Elm Tailwind CSS Template

- Elm + Tailwind CSS + Webpack
- Supports `wails dev` hot reloading

Wails is awesome, thanks for sharing such a great tool!

https://github.com/rnice01/wails-elm-tailwind-template

**Reply by misitebao** [reply]
It is recommended to use the naming format of `wails-template-xxx`

**Comment by xijaja** [comment]
### Vite Solid template

https://github.com/xijaja/wails-template-solid-js

https://github.com/xijaja/wails-template-solid-ts

**Comment by plihelix** [comment]
## Minimally invasive! Clean SvelteKit template.
#### & How-to use any SvelteKit npm template.

* Uses `@sveltejs/adapter-static` to build the front end into `/frontend/build` for embedding into the wails app.

[plihelix/wails-template-sveltekit](https://github.com/plihelix/wails-template-sveltekit)

Bonus: [simple bash script that builds a wails project and installs a [SkeletonUI](https://www.skeleton.dev) front end which uses SvelteKit & TailwindCSS can be found at [plihelix/wails-skeletonui-script](https://github.com/plihelix/wails-skeletonui-script).

(photo of the script result not the clean SvelteKit template)
![image](https://user-images.githubusercontent.com/73962193/211171657-c4489f4f-0842-42c1-b041-e187c2b8a48e.png)

**Reply by leaanthony** [reply]
This is awesome! Love Skeleton!

**Reply by plihelix** [reply]
I've adjusted the name of the repo to conform to the template recommendations.

**Comment by robin-samuel** [comment]
### Wails SvelteKit Skeleton Template + TailwindCSS

https://github.com/robin-samuel/wails-sveltekit-skeleton-template

**Reply by leaanthony** [reply]
Awesome!

**Comment by haukened** [comment]
## Wails SvelteKit + Typescript

https://github.com/haukened/wails-sveltekit-ts

includes a lot of the "gotchas" in sveltekit setup like SPA mode, build folders, etc thanks to @figuerom16

**Reply by haukened** [reply]
I'm also considering forking this to include systray support for v2 sites, as thats what i do with a lot of my projects. If anyone's interested i'll throw one up with sveltekit + tailwindcss + daisyUI + systray

**Comment by PylotLight** [comment]
### Wails Svelte + Tailwind
---
https://github.com/PylotLight/wails-vite-svelte-tailwind-template
Previous similar template was out of date for various packages and wails version changes so updated all backend/front libs + updated the sample app.svelte to use some more styling to both remove the blinding white page and demo usage of tailwind.

**Comment by LazyClicks** [comment]
### Wails Qwik template
https://github.com/LazyClicks/wails-qwik-template

**Comment by vixalie** [comment]
# React Template using Vite

Simplest Wails 2.x template with React and Javascript, using Vite for asset building.

Included:

- Build tools: Vite 4.4
- Framework: React 18.2

Link: [wails+vite+react](https://github.com/vixalie/wails-vite-react)

**Comment by id88** [comment]
# wails-vite-vue-ts-navie-pinia
A template for Wails with Vite + Vue + TypeScript + NavieUI + Pinia. 

 - Wails + Vite4 + Vue3+ Typescript5 + Pinia2 + Navie-UI 
- Support hot reload with `wails dev`.
- Automatically built using GitHub action (See: [Github Action](https://github.com/id88/wails-vite-vue-ts-navie-pinia/actions))

Link:
https://github.com/id88/wails-vite-vue-ts-navie-pinia

![screenshot](https://raw.githubusercontent.com/id88/wails-vite-vue-ts-navie-pinia/main/screenshot.png)

**Comment by PylotLight** [comment]
# wails-htmx-templ-chi-tailwind
A template for Wails with HTMX + Templ + Chi + Vite + Tailwind + DaisyUI.
Support hot reload with wails dev.

Link:
https://github.com/PylotLight/wails-hmtx-templ-template
PR:
https://github.com/wailsapp/wails/pull/2984

**Comment by 9dl** [comment]
NextJS 14.0.3 (No TypeScript) with Tailwind CSS
 > NextJS project without TypeScript, equipped with the latest NextJS version and Tailwind CSS preinstalled.
https://github.com/9dl/Wails-NextJS-Template

**Comment by mJehanno** [comment]
# Wails-template-angular-latest

A template that will try its best to support angular latest version. It will probably be easier  to use wails and angular together as angular now use vitejs under the hood !

In fact I didn't even needed to add [angular specific stuff](https://wails.io/docs/guides/angular) in wails configuration to make it work.

![Snapshot_2023-12-16_08-43-29](https://github.com/wailsapp/wails/assets/9280964/34c4b99f-9187-4288-9542-c70a803b9e35)


link : https://github.com/mJehanno/wails-template-angular-latest

pr: #3135

**Comment by Braincompiler** [comment]
# Wails + Lit + Shoelace + esbuild + Typescript

* Incl. pre-configured prettier and Typescript
* Supports 🔥hot-reload via `wails dev` (and faster hot replacement for CSS without reloading)

Usage:
```shell
wails init -n my-app -t https://github.com/Braincompiler/wails-lit-shoelace-esbuild-template
```

Link:
https://github.com/Braincompiler/wails-lit-shoelace-esbuild-template

**Comment by Gornius** [comment]
# Wails + Nuxt3 + Typescript

* Almost vanilla Nuxt3 starter template
* Auto-import for wails js runtime included

Usage:
```
wails init -n "Your Project Name" -t https://github.com/gornius/wails-template-nuxt
```
![image](https://github.com/wailsapp/wails/assets/34241360/d543d4e6-0ae3-44a4-b05b-524a6a401175)


Link: https://github.com/Gornius/wails-template-nuxt

**Comment by kairo913** [comment]
# Wails + Next.js +TailwindCSS

A template for Wails with Next.js v15 and TailwindCSS v4.
Support hot reload with wails dev.

Usage:
```
wails init -n "Your Project Name" -t https://github.com/kairo913/wails-nextjs-tailwind-template
```

![wails-nextjs-tailwind-template](https://github.com/user-attachments/assets/57d786e2-693c-4a39-a027-27c31428409b)

Link: https://github.com/kairo913/wails-nextjs-tailwind-template

**Comment by xijaja** [comment]
I created a new template that uses sveltekit and shadcn-svelte；

![ui-1](https://github.com/user-attachments/assets/7e5a5833-0dbe-4035-9edc-72921e629ba7)
![ui-2](https://github.com/user-attachments/assets/81f47686-f493-40d6-a7fb-09ecd0c7fa35)

Link：https://github.com/xijaja/wails-template-shadcn-svelte

**Comment by ericnsmith** [comment]
Created a new htmx / templ [template](https://gitlab.com/ericnsmith/htmx-templ-wails-template).

Instead of the usual HTMX approach of using endpoint url methods (hx-get, hx-post, etc.) for data retrieval, it uses the awesome [wails-htmx](https://github.com/lllama/wails-htmx) HTMX extension which allows binding to Wails javascript runtime methods.

**Comment by TekWizely** [comment]
New template that uses **Vue** + **PrimeVue** + PrimeVue's **Sakai** Application Template:

[TekWizely/wails-template-primevue-sakai](https://github.com/TekWizely/wails-template-primevue-sakai)

![Screenshot](https://raw.githubusercontent.com/TekWizely/wails-template-primevue-sakai/main/.github/screenshot.png)

**Comment by Alex6357** [comment]
# Wails + Svelte + TypeScript + less + Prettier + ESLint + Vite

[https://github.com/Alex6357/wails-template-svelte-ts-less-prettier-eslint-vite](https://github.com/Alex6357/wails-template-svelte-ts-less-prettier-eslint-vite)

The template is based on the official svelte-ts template, upgraded to Svelte 5, manually installed less, Prettier and ESLint. The configuration files are from official SvelteKit template created using `npx sv create`.

**Comment by Alex6357** [comment]
# Wails + SvelteKit + less + Prettier + ESLint

[https://github.com/Alex6357/wails-template-sveltekit-less-prettier-eslint](https://github.com/Alex6357/wails-template-sveltekit-less-prettier-eslint)

The frontend is generated using `npx sv create` and manually installed less.

**Comment by poisonshell** [comment]
**Wails +  React + TypeScript + TailwindCSS, + TanStack Router  Template**


<img width="1193" height="773" alt="Screenshot 2025-09-02 at 02 45 08" src="https://github.com/user-attachments/assets/1f42a1ae-5ef9-4953-9dbc-7e1c9dabb900" />

https://github.com/poisonshell/wails-vite-react-ts

A Wails desktop application template with React, TypeScript, TailwindCSS, and TanStack Router.

- Wails 2.10.2 
- React 19 with TypeScript 5.7
- Vite 7
- TailwindCSS 4 
- TanStack Router 1.131

**Comment by Mahcks** [comment]
Hey everyone! 👋

I've created a modern Wails template that I think will be a nice base for a lot of projects :)

## 🚀 Features

- **Wails v2.11.0** - Latest version with all improvements
- **React 18.3 + TypeScript 5.7** - Modern React with full type safety
- **Vite 5.4** - Lightning-fast HMR and dev experience
- **Tailwind CSS v4** - Brand new version with the `@tailwindcss/vite` plugin
- **shadcn/ui** - Beautiful, accessible components ready to use
- **Cross-platform build scripts** - Easy builds for Windows, macOS, and Linux

## 📦 Quick Start

```bash
wails init -n myapp -t https://github.com/Mahcks/wails-vite-react-tailwind-shadcnui-ts
cd myapp
wails dev
```
**https://github.com/Mahcks/wails-vite-react-tailwind-shadcnui-ts**

**Comment by JinGongX** [comment]
### Wails + Vue + Typescript + Tailwind CSS (for Wails v3) template
🌍 A Wails v3-based desktop application template with i18n, dark mode, and SQLite integration.
﻿
✅ Internationalization (i18n) using vue-i18n
🌗 Automatic light/dark theme support with manual override
🗃️ SQLite database integration with CRUD operations
⚙️ Settings panel and hotkey configuration demo
💡 Built with Wails v3, Vue 3, and TypeScript
🚀 `wails3 init -n myproject -t https://github.com/JinGongX/SuiDemo` Create a project

https://github.com/JinGongX/SuiDemo

**Comment by ErsinCabuk** [comment]
# 🚀 New Template Announcement: Wails v2 + Angular v21 + TailwindCSS v4 + Bun

Hello everyone! 👋

I've put together a modern, fast, and lightweight starter template for anyone who wants to build Wails v2 applications using the latest Angular ecosystem. I've also just submitted a PR to the official Wails repository to get this added to the community templates!

## ✨ What's inside?
This template brings together the powerful Go-based backend of Wails with Angular's innovative features:
* **Angular v21:** Uses the latest standalone architecture, Control Flow, and Signals out of the box.
* **TailwindCSS v4:** Configured for rapid and modern UI development.
* **Bun Package Manager:** Replaces `npm`/`yarn` for lightning-fast frontend dependency management and dev server execution (`bunx ng serve`).
* **Clean & Minimal:** Includes a fully functioning, minimal base project demonstrating the Go to Angular interop (the classic `Greet` function).

## 📦 Quick Start
You can test it out right now without waiting for the PR to be merged! Just run:

```bash
wails init -n my-project -t https://github.com/ErsinCabuk/wails-angular-template

**Comment by Jemeni11** [comment]
# ReactViteTSWails. Wails Template. React v19 + Vite v8 + TypeScript v6 + TailwindCSS + Prettier + ESLint + Stylelint + lint-staged + husky

## Introduction

Hey. I'm a front-end developer trying to learn Golang. I found out about Wails, and I'm planning
on using it as my daily driver for building desktop apps. I already have a template that I use for my React + Vite projects called [ReactViteTS](https://github.com/Jemeni11/ReactViteTS), so I decided to create a Wails template based on that.

I don't know much Golang yet, so this template doesn't really have much going on there except for the basic Wails app structure. The frontend, however, is fully set up with the latest versions of React, Vite, and TypeScript, along with TailwindCSS, Prettier, ESLint, Stylelint and lint-staged. The React Compiler is also configured.

## Quick Start

```bash
wails init -n my-project -t https://github.com/Jemeni11/ReactViteTSWails
cd my-project
wails dev
```

https://github.com/Jemeni11/ReactViteTSWails

Happy coding 🚀
